io/net/uperf_test.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2017 IBM
+# Author: Harsha Thygaraaja <harshkid@linux.vnet.ibm.com>
+
+"""
+Unified Performance Tool or uperf for short, is a network
+performance measurement tool that supports execution of
+workload profiles
+"""
+
+import os
+import time
+import netifaces
+from avocado import main
+from avocado import Test
+from avocado.utils.software_manager import SoftwareManager
+from avocado.utils import distro
+from avocado.utils import build
+from avocado.utils import archive
+from avocado.utils import process
+from avocado.utils.genio import read_file
+
+
+class Uperf(Test):
+    """
+    Uperf Test
+    """
+
+    def setUp(self):
+        """
+        To check and install dependencies for the test
+        """
+        smm = SoftwareManager()
+        detected_distro = distro.detect()
+        pkgs = ["gcc", "autoconf", "perl", "m4", "git-core"]
+        if detected_distro.name == "Ubuntu":
+            pkgs.extend(["libsctp1", "libsctp-dev", "lksctp-tools"])
+        else:
+            pkgs.extend(["lksctp-tools", "lksctp-tools-devel"])
+        for pkg in pkgs:
+            if not smm.check_installed(pkg) and not smm.install(pkg):
+                self.cancel("%s package is need to test" % pkg)
+        interfaces = netifaces.interfaces()
+        self.iface = self.params.get("interface", default="")
+        self.peer_ip = self.params.get("peer_ip", default="")
+        if self.iface not in interfaces:
+            self.cancel("%s interface is not available" % self.iface)
+        if self.peer_ip == "":
+            self.cancel("%s peer machine is not available" % self.peer_ip)
+        self.peer_user = self.params.get("peer_user_name", default="root")
+        uperf_download = self.params.get("uperf_download", default="https:"
+                                         "//github.com/uperf/uperf/"
+                                         "archive/master.zip")
+        tarball = self.fetch_asset(uperf_download, expire='7d')
+        archive.extract(tarball, self.teststmpdir)
+        self.uperf_dir = os.path.join(self.teststmpdir, "uperf-master")
+        cmd = "scp -r %s %s@%s:/tmp" % (self.uperf_dir, self.peer_user,
+                                        self.peer_ip)
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
+            self.cancel("unable to copy the uperf into peer machine")
+        tmp = "cd /tmp/uperf-master;./configure ppc64le;make"
```

Comment:
I think no need to use tmp here, you can use cmd only.

Reply:
This is purely cosmetic. Rather than making the command line seem longer, I'd rather use variable "tmp".

Reply:
@hathyaga What @vrbagalkote meant was, use `cmd` instead of `tmp`